config_parser_test.cc

```diff
@@ -51,10 +51,54 @@ TEST_F(NginxStringConfigTest, MismatchedBraces) {
 	ASSERT_FALSE(ParseString("server { listen 80; something { some statement; } "));
 }
 
+// Test Basic ToString
+TEST_F(NginxStringConfigTest, ToString) {
+	ASSERT_TRUE(ParseString("foo bar;"));
+	EXPECT_EQ("foo bar;\n", out_config_.ToString(0));
+}
+
+// Test Nested ToString
+TEST_F(NginxStringConfigTest, NestedToString) {
+	ASSERT_TRUE(ParseString("parent{ foo \"bar\"; child { statement; }}"));
+	EXPECT_EQ("parent {\n  foo \"bar\";\n  child {\n    statement;\n  }\n}\n", out_config_.ToString(0));
+}
+
+// Test Empty Config
+TEST_F(NginxStringConfigTest, EmptyConfig) {
+	ASSERT_FALSE(ParseString(" "));
+}
+
+// Test Mismatched single quotes
+TEST_F(NginxStringConfigTest, MismatchedQuotes) {
+	ASSERT_FALSE(ParseString(" 'foo "));
+}
+
+// Test Open bracket after semicolon
+TEST_F(NginxStringConfigTest, BadBracket) {
+	ASSERT_FALSE(ParseString(" foo bar; { statement; } "));
+}
+
+// Test Bad bracket (missing semicolon)
+TEST_F(NginxStringConfigTest, BadBracketMissingSemicolon) {
+	ASSERT_FALSE(ParseString(" server { statement } "));
+}
+
+// Test Bad semicolon
+TEST_F(NginxStringConfigTest, BadSemicolon) {
+	ASSERT_FALSE(ParseString(" server { statement; }; "));
+}
+
 //Checks that a large config is parsed correctly
 //Code taken from https://www.nginx.com/resources/wiki/start/topics/examples/full/
 TEST(NginxConfigParserTest, FullExampleConfig) {
 	NginxConfigParser parser;
 	NginxConfig out_config_;
 	EXPECT_TRUE(parser.Parse("fullex_config", &out_config_));
 }
+
+// Test bad config file
+TEST(NginxConfigParserTest, BadConfig) {
```

Comment:
I have a test fixture since we are testing more than one config.

//Creates test fixture for Parse (file name input)
class NginxConfigParserTest:public testing::Test {
	protected:
		bool ParseFile(const char* file_name) {
			std::ifstream config_file;
			config_file.open(file_name);
			bool ret = parser.Parse(dynamic_cast<std::istream*>(&config_file), &out_config);
			config_file.close();
			return ret;
		}
	NginxConfigParser parser;
	NginxConfig out_config;
};

Dunno if we need it.